firestore/query.go

```diff
@@ -1036,3 +1045,72 @@ func (it *btreeDocumentIterator) next() (*DocumentSnapshot, error) {
 }
 
 func (*btreeDocumentIterator) stop() {}
+
+// AggregationQuery allows for generating aggregation results of an underlying
+// basic query. A single AggregationQuery can contain multiple aggregations.
+type AggregationQuery struct {
+	aggregateQueries []*pb.StructuredAggregationQuery_Aggregation // aggregateQueries contains all of the queries for this request.
+	query            *Query                                       // query contains a reference pointer to the underlying structured query.
+}
+
+// WithCount specifies that the aggregation query provide a count of results
+// returned by the underlying Query.
+func (a *AggregationQuery) WithCount(alias string) *AggregationQuery {
+	aq := &pb.StructuredAggregationQuery_Aggregation{
+		Alias:    alias,
+		Operator: &pb.StructuredAggregationQuery_Aggregation_Count_{},
+	}
+
+	a.aggregateQueries = append(a.aggregateQueries, aq)
+
+	return a
+}
+
+// Get retrieves the aggregation query results from the service.
+func (a *AggregationQuery) Get(ctx context.Context) (AggregationResult, error) {
```

Comment:
I see we use `map[string]interface{}` elsewhere. It's probably best to match that an not introduce a type alias for a map?

https://pkg.go.dev/cloud.google.com/go/firestore#DocumentSnapshot.Data

Reply:
Ack. Resolved over chat.